examples/priviledges/main.tf

```diff
@@ -0,0 +1,192 @@
+#        | { schemaObjectPrivileges | ALL [ PRIVILEGES ] } ON { <object_type> <object_name> | ALL <object_type>S IN SCHEMA <schema_name> }  }
+#        | { schemaObjectPrivileges | ALL [ PRIVILEGES ] } ON FUTURE <object_type>S IN SCHEMA <schema_name> }  }
+#   TO [ ROLE ] <role_name> [ WITH GRANT OPTION ]
+
+# globalPrivileges ::=
+#   { { CREATE { ROLE | USER | WAREHOUSE | DATABASE } } | MANAGE GRANTS | MONITOR USAGE } [ , ... ] }
+
+# accountObjectPrivileges ::=
+# -- For RESOURCE MONITOR
+#   { MODIFY | MONITOR } [ , ... ]
+# -- For WAREHOUSE
+#   { MODIFY | MONITOR | USAGE | OPERATE } [ , ... ]
+# -- For DATABASE
+#   { MODIFY | MONITOR | USAGE | CREATE SCHEMA | IMPORTED PRIVILEGES } [ , ... ]
+
+# schemaPrivileges ::=
+#   { MODIFY | MONITOR | USAGE | CREATE { TABLE | VIEW | FILE FORMAT | STAGE | PIPE | SEQUENCE | FUNCTION } } [ , ... ]
+
+# schemaObjectPrivileges ::=
+# -- For TABLE
+#   { SELECT | INSERT | UPDATE | DELETE | TRUNCATE | REFERENCES } [ , ... ]
+# -- For VIEW
+#     SELECT
+# -- For internal STAGE
+#     READ [ , WRITE ]
+# -- For external STAGE
+#     USAGE
+# -- For FILE FORMAT, UDF, or SEQUENCE
+#     USAGE
+
+# Given that schema, here are some terraform resources we could support. We should map this out semi-fully, even though
+# we will build incrementally.
+
+# set global privs
+resource "snowflake_priviledge_grants" "a" {
+  name      = "foo1"
+  role_name = "role1"
+
+  global {
+    all = true
+
+    # these two ^ v are mutually excluseive
+    # priviledges = ["create_role", "create_user", "create_warehouse", "create_database", "manage_grants", "monitor_usage"]
+    # any subset of these can be supplied ^
+  }
+}
+
+# set account object privs
+
+# resource monitor
+resource "snowflake_priviledge_grants" "b" {
+  name      = "foo2"
+  role_name = "role2"
+
+  resource_monitor {
+    name        = "bar"                 // name of the resource monitor
+    priviledges = ["modify", "monitor"]
+  }
+}
+
+# warehouse
+resource "snowflake_priviledge_grants" "c" {
+  name      = "foo3"
+  role_name = "role3"
+
+  warehouse {
+    name        = "bar"                                     // name of the warehouse
+    priviledges = ["modify", "monitor", "usage", "operate"]
+  }
+}
+
+# database
+resource "snowflake_priviledge_grants" "d" {
+  name      = "foo4"
+  role_name = "role4"
+
+  database {
+    name        = "bar"                                                                   // name of the database
+    priviledges = ["modify", "monitor", "usage", "create_schema", "imported_priviledges"]
+  }
+}
+
+# database
+resource "snowflake_priviledge_grants" "e" {
+  name      = "foo5"
+  role_name = "role5"
+
+  schema {
+    schema_name = "bar"
+
+    # these two ^ v are mutually exclusive
+    # database_name = "bar"
+    #  if ^ then it means "all schemas in db"
+
+    # not sure if we should make these mutually exclusive or allow you to create one resource w/ a bunch of privs
+    table {
```

Comment:
not sure if you can have nested blocks like these in the current version of tf

Reply:
i see an example here https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html

Reply:
nvm, you can